src/components/navigation/perPage/NavigationPerPage.tsx

```diff
@@ -41,31 +44,23 @@ export class NavigationPerPage extends React.Component<INavigationPerPageProps,
     }
 
     componentWillMount() {
-        this.perPageNumbers = this.props.perPageNumbers || PER_PAGE_NUMBERS;
-        if (this.props.onRender) {
-            this.props.onRender(this.perPageNumbers[this.props.initialPosition]);
-        }
+        this.initialPosition = !_.isUndefined(this.props.initialPosition) ? this.props.initialPosition : Math.ceil(this.props.perPageNumbers.length / 2) - 1;
+        callIfDefined(this.props.onRender, this.props.perPageNumbers[this.initialPosition]);
     }
 
     componentWillUnmount() {
-        if (this.props.onDestroy) {
-            this.props.onDestroy();
-        }
+        callIfDefined(this.props.onDestroy);
     }
 
     render() {
-        this.perPageNumbers = this.props.perPageNumbers || PER_PAGE_NUMBERS;
-
-        const currentPerPage: number = this.props.currentPerPage || this.perPageNumbers[this.props.initialPosition];
+        const currentPerPage: number = this.props.currentPerPage || this.props.perPageNumbers[this.initialPosition];
         const topNumber: number = this.props.totalEntries;
-        const label: string = this.props.label || PER_PAGE_LABEL;
 
-        const perPageSelects: JSX.Element[] = _.map(this.perPageNumbers, (num: number, index: number): JSX.Element => {
-            const shouldShowPerPageSelect: boolean =
-                topNumber > (this.perPageNumbers[index - 1] || 0);
+        const perPageSelects: JSX.Element[] = _.map(this.props.perPageNumbers, (num: number, index: number): JSX.Element => {
+            const shouldShowPerPageSelect: boolean = topNumber > (this.props.perPageNumbers[index - 1] || 0);
 
             if (shouldShowPerPageSelect) {
-                const selectId: string = 'perpage-' + (this.props.id || '') + num;
+                const selectId: string = `perpage-${this.props.id || ''}-${num}`;

```

Comment:
The ID now has an additional `-` before `num`, intended? 

It's only used as `key` so it should not be breaking, but just to be sure :)

Reply:
It's intended. I felt it would be cleaner and as you said, it's only used as a key